src/components/WebCLI/WebCLI.tsx

```diff
@@ -132,17 +138,32 @@ const WebCLI = ({
       onclose: () => {
         // Unused handler.
       },
+      onerror: (error) => {
+        // Only display errors if they're related to the current WebSocket
+        // connection.
+        if (connection.current?.isWebSocketEqual(conn)) {
+          setInlineError(
+            InlineErrors.CONNECTION,
+            typeof error === "string" ? error : Label.UNKNOWN_ERROR,
+          );
+        }
+      },
       messageCallback: (message: string) => {
         wsMessageStore.current = wsMessageStore.current + message;
         setOutput(wsMessageStore.current);
       },
     }).connect();
-    setConnection(conn);
-    return () => {
-      conn.disconnect();
-    };
+    connection.current?.disconnect();
```

Comment:
Do we need this disconnect here? It checks that there isn't an active connection at the top of this useEffect.

Reply:
You're right! Thanks for spotting this. Removed it.